app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -3,8 +3,51 @@ kind: registration
 name: Global Registration
 model: ProvisioningTemplate
 -%>
+<%
+    curl_opts = ["-H 'Content-Type: application/json'", "-H 'Accept: application/json'", "-H 'Authorization: Bearer #{@auth_token}'"]
+%>
+#!/bin/bash
+
+# Template parameters:
+<%= "#  - User: [#{@user.login}]" if @user %>
+<%= "#  - Organization: [#{@organization.name}]" if @organization %>
+<%= "#  - Location: [#{@location.name}]" if @location %>
+<%= "#  - Hostgroup: [#{@hostgroup.name}]" if @hostgroup %>
 
 if ! [ $(id -u) = 0 ]; then
     echo "Please run as root"
     exit 1
 fi
+
+if [ -f /etc/os-release ] ; then
+  . /etc/os-release
+fi
+
+HOSTNAME=$(hostname -f)
+
+echo "#"
+echo "# Running registration"
+echo "# Hostname: $HOSTNAME"
+echo "# OS: $NAME"
+echo "#"
+
+if [ $ID = rhel ] ; then
+    echo "RHEL registration"
+<% if plugin_present?('katello') %>
+    CONSUMER_RPM=$(mktemp --suffix .rpm)
+
+    curl --insecure --output $CONSUMER_RPM <%= subscription_manager_configuration_url %>
+    yum localinstall $CONSUMER_RPM -y
+    yum install subscription-manager -y
+    subscription-manager register --org=<%= @organization.label %> --activationkey="sample-key"
```

Comment:
Hardcoded activation key?

Reply:
For now, yes, because we don't pass :actiovation_key param to template.
GRT endpoint is going to be extended from katello plugin - [See #30642](https://projects.theforeman.org/issues/30642)